packages/ui-pattern-block/src/UIPatternBlock.tsx

```diff
@@ -1,25 +1,238 @@
+        cssToInject,
+        npmDependencies,
+        externalDependencies,
+        reset,
+    ]
+        .map((item) => item.toString())
+        .join('');
 
     return (
-        <div className="ui-pattern-block">
-            <Sandpack
-                template={sandpackTemplate as SandpackPredefinedTemplate}
-                theme={sandpackThemeValues[sandpackTheme]}
+        <div key={sandpackTemplate} data-test-id="ui-pattern-block" className="ui-pattern-block">
+            <Captions
+                appBridge={appBridge}
+                title={title}
+                description={description}
+                onTitleChange={(newValue) => setBlockSettings({ title: newValue })}
+                onDescriptionChange={(newValue) => setBlockSettings({ description: newValue })}
+                isEditing={isEditing}
             />
+            <div
+                data-test-id="ui-pattern-block-wrapper"
+                style={{
+                    ...(hasBorder && getBorderStyles(borderStyle, borderWidth, borderColor)),
+                    borderRadius: getRadiusStyle(hasRadius, radiusValue, radiusChoice),
+                }}
+                className="tw-rounded tw-bg-white tw-overflow-hidden"
+            >
+                <SandpackProvider
+                    files={templateFiles}
+                    template={sandpackTemplate}
+                    customSetup={parsedNpmDependencies}
+                    theme={sandpackThemeValues[sandpackTheme]}
+                    key={sandpackRestartInitiators}
+                    options={{
+                        classes: EDITOR_CLASSES,
+                        activeFile: initialActiveFile[sandpackTemplate],
+                        externalResources: [cssToInject, ...parsedExternalDependencies],
+                    }}
+                >
+                    <SandpackLayout className="tw-flex tw-flex-col tw-bg-inher">
+                        {isResponsivePreviewOpen && (
+                            <ResponsivePreview onClose={() => setIsResponsivePreviewOpen(false)} />
+                        )}
+                        <SandpackPreview
+                            style={{
+                                height: getHeightStyle(isCustomHeight, customHeightValue, heightChoice),
+                                padding: getPaddingStyle(hasCustomPadding, paddingCustom, paddingChoice),
+                                ...(hasBackground && {
+                                    ...getBackgroundColorStyles(backgroundColor),
+                                    backgroundImage: 'none',
+                                }),
+                            }}
+                            showRefreshButton={false}
+                            showOpenInCodeSandbox={false}
+                            className="tw-rounded-none tw-bg-box-nega tw-shadow-none tw-ml-0"
+                        />
+                        {(isEditing || showCode) && (
+                            <CodeEditor
```

Comment:
Can't access the CodeEditor by keyboard

Reply:
Fixed.

Reply:
With the escape key. Unfortunately the content manager also listens to escape and unfocuses the block.

This PR will fix that: https://github.com/Frontify/web-app/pull/1542

Reply:
Works as expected with the open PR